eth/api.go

```diff
@@ -441,8 +461,19 @@ func formatError(err error) string {
 
 // TraceTransaction returns the structured logs created during the execution of EVM
 // and returns them as a JSON object.
-func (api *PrivateDebugAPI) TraceTransaction(txHash common.Hash, logConfig *vm.LogConfig) (*ethapi.ExecutionResult, error) {
-	logger := vm.NewStructLogger(logConfig)
+func (api *PrivateDebugAPI) TraceTransaction(txHash common.Hash, config *TraceArgs) (interface{}, error) {
+	var tracer vm.Tracer
+	if config != nil && config.tracer != "" {
+		var err error
+		tracer, err = ethapi.NewJavascriptTracer(config.tracer)
```

Comment:
You can also make this a one liner:

`if tracer, err = ethapi.NewJavascriptTracer(config.tracer); err != nil {`